jgiven-core/src/main/java/com/tngtech/jgiven/impl/util/ReflectionUtil.java

```diff
@@ -180,109 +164,134 @@ public static void invokeMethod( Object object, Method method, String errorDescr
     }
 
-    public static List<String> getAllFieldNames( Iterable<Field> fields ){
-        return StreamSupport.stream( fields.spliterator(), false )
-                .map( Field::getName )
-                .collect( Collectors.toList() );
+    public static List<String> getAllFieldNames(Iterable<Field> fields) {
+        return StreamSupport.stream(fields.spliterator(), false)
+                .map(Field::getName)
+                .collect(Collectors.toList());
     }
 
-    public static void setField( Field field, Object object, Object value, String errorDescription ){
-        makeAccessible( field, errorDescription );
+    public static void setField(Field field, Object object, Object value, String errorDescription) {
+        makeAccessible(field, errorDescription);
         try {
-            field.set( object, value );
-        } catch( IllegalArgumentException e ) {
-            log.debug( "Caught exception:", e );
-            throw new JGivenInjectionException( "Could not set " + toReadableString( field ) + errorDescription +
-                    " to value " + value + ": " + e.getMessage(), e );
-        } catch( IllegalAccessException e ) {
-            log.debug( "Caught exception:", e );
-            throw new JGivenInjectionException( "Could not set " + toReadableString( field ) + errorDescription +
+            field.set(object, value);
+        } catch (IllegalArgumentException e) {
+            log.debug("Caught exception:", e);
+            throw new JGivenInjectionException("Could not set " + toReadableString(field) + errorDescription +
+                    " to value " + value + ": " + e.getMessage(), e);
+        } catch (IllegalAccessException e) {
+            log.debug("Caught exception:", e);
+            throw new JGivenInjectionException("Could not set " + toReadableString(field) + errorDescription +
                     ", because of insuffient access rights. "
-                    + "Either make the field public or disable your security manager while executing JGiven tests.", e );
+                    + "Either make the field public or disable your security manager while executing JGiven tests.", e);
         }
     }
 
-    public static void makeAccessible( AccessibleObject object, String errorDescription ){
+    public static void setField(String fieldName, Object object, Object value, String errorDescription) {
         try {
-            object.setAccessible( true );
-        } catch( SecurityException e ) {
-            log.debug( "Caught exception: ", e );
-            log.warn( "Could not make {} accessible, trying to access it nevertheless and hoping for the best.",
-                    toReadableString( object ), errorDescription );
+            setField(object.getClass().getDeclaredField(fieldName), object, value, errorDescription);
+        } catch (NoSuchFieldException e) {
+            log.debug(
+                    format("Not able to access field '%s'." + errorDescription, fieldName), e);
```

Comment:
also I am not certain that passing an error description is worth it here. You are probably fine with 
```
("Not able to access field '{}' of class '{}'", fieldName, object.getClass().getSimpleName())
```